failed: ClusterClientSpec with Artery

https://jenkins.akka.io:8498/job/akka-artery-cluster-tests/57/consoleText

```
[JVM-1] - must re-establish connection to receptionist after server restart *** FAILED ***
[JVM-1]   java.lang.AssertionError: timeout (10 seconds) waiting for 1 messages on InfoFilter(Some(akka://ClusterClientSpec@a0.moxie:42157/user/client4),Left(Connected to),false)
[JVM-1]   at akka.testkit.EventFilter.intercept(TestEventListener.scala:114)
[JVM-1]   at akka.cluster.client.ClusterClientSpec$$anonfun$1$$anonfun$apply$mcV$sp$32$$anonfun$apply$mcV$sp$33$$anonfun$apply$mcV$sp$34.apply$mcV$sp(ClusterClientSpec.scala:417)
[JVM-1]   at akka.remote.testkit.MultiNodeSpec.runOn(MultiNodeSpec.scala:341)
[JVM-1]   at akka.cluster.client.ClusterClientSpec$$anonfun$1$$anonfun$apply$mcV$sp$32$$anonfun$apply$mcV$sp$33.apply$mcV$sp(ClusterClientSpec.scala:404)
[JVM-1]   at akka.cluster.client.ClusterClientSpec$$anonfun$1$$anonfun$apply$mcV$sp$32$$anonfun$apply$mcV$sp$33.apply(ClusterClientSpec.scala:403)
[JVM-1]   at akka.cluster.client.ClusterClientSpec$$anonfun$1$$anonfun$apply$mcV$sp$32$$anonfun$apply$mcV$sp$33.apply(ClusterClientSpec.scala:403)
```

--- a/akka-cluster-tools/src/multi-jvm/scala/akka/cluster/client/ClusterClientSpec.scala
+++ b/akka-cluster-tools/src/multi-jvm/scala/akka/cluster/client/ClusterClientSpec.scala
@@ -431,10 +431,13 @@ class ClusterClientSpec extends MultiNodeSpec(ClusterClientSpec) with STMultiNod
       runOn(remainingServerRoleNames.toSeq: _*) {
         Await.ready(system.whenTerminated, 20.seconds)
         // start new system on same port
+        val port = Cluster(system).selfAddress.port.get
         val sys2 = ActorSystem(
           system.name,
-          ConfigFactory.parseString("akka.remote.netty.tcp.port=" + Cluster(system).selfAddress.port.get)
-            .withFallback(system.settings.config))
+          ConfigFactory.parseString(
+            if (RARP(system).provider.remoteSettings.Artery.Enabled) s"akka.remote.artery.canonical.port=$port"
+            else s"akka.remote.netty.tcp.port=$port"
+          ).withFallback(system.settings.config))
         Cluster(sys2).join(Cluster(sys2).selfAddress)
         val service2 = sys2.actorOf(Props(classOf[TestService], testActor), "service2")
         ClusterClientReceptionist(sys2).registerService(service2)

